### PR TITLE
gh-103527: Add multibytecodec.h as make dep for _codecs_*

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-103527

### CODE REVIEW

File: Makefile.pre.in [hunk 1]
```diff
@@ -2662,12 +2662,12 @@ MODULE__SOCKET_DEPS=$(srcdir)/Modules/socketmodule.h $(srcdir)/Modules/addrinfo.
...
+MODULE__CODECS_CN_DEPS=$(srcdir)/Modules/cjkcodecs/mappings_cn.h $(srcdir)/Modules/cjkcodecs/cjkcodecs.h $(srcdir)/Modules/cjkcodecs/multibytecodec.h
+MODULE__CODECS_HK_DEPS=$(srcdir)/Modules/cjkcodecs/mappings_hk.h $(srcdir)/Modules/cjkcodecs/cjkcodecs.h $(srcdir)/Modules/cjkcodecs/multibytecodec.h
+MODULE__CODECS_ISO2022_DEPS=$(srcdir)/Modules/cjkcodecs/mappings_jisx0213_pair.h $(srcdir)/Modules/cjkcodecs/cjkcodecs.h $(srcdir)/Modules/cjkcodecs/alg_jisx0201.h $(srcdir)/Modules/cjkcodecs/emu_jisx0213_2000.h $(srcdir)/Modules/cjkcodecs/multibytecodec.h
+MODULE__CODECS_JP_DEPS=$(srcdir)/Modules/cjkcodecs/mappings_jisx0213_pair.h $(srcdir)/Modules/cjkcodecs/cjkcodecs.h $(srcdir)/Modules/cjkcodecs/alg_jisx0201.h $(srcdir)/Modules/cjkcodecs/emu_jisx0213_2000.h $(srcdir)/Modules/cjkcodecs/mappings_jp.h $(srcdir)/Modules/cjkcodecs/multibytecodec.h
+MODULE__CODECS_KR_DEPS=$(srcdir)/Modules/cjkcodecs/mappings_kr.h $(srcdir)/Modules/cjkcodecs/cjkcodecs.h $(srcdir)/Modules/cjkcodecs/multibytecodec.h
+MODULE__CODECS_TW_DEPS=$(srcdir)/Modules/cjkcodecs/mappings_tw.h $(srcdir)/Modules/cjkcodecs/cjkcodecs.h $(srcdir)/Modules/cjkcodecs/multibytecodec.h
```
Comment (zware) [inline Makefile.pre.in]: Could some of this be simplified to `$(srcdir)/Modules/cjkcodecs/*.h`? Or at least it looks like `cjkcodecs.h` and `multibytecodecs.h` are common to all and could be referred to by a separate variable?

Comment (erlend-aasland) [reply Makefile.pre.in]: Yeah, a variables makes sense. I'm not a big fan of these long dep lines anyway.

Comment (erlend-aasland) [reply Makefile.pre.in]: I put `cjkcodecs.h` and `multibytecodec.h` in a variable named `CODECS_COMMON_HEADERS`. It's a lot cleaner now; thanks.

**Review by zware — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks for the review Zach!

**Pull request merged**